addons/test_mail/tests/test_mail_channel.py

```diff
@@ -274,6 +274,87 @@ def test_multi_company_chat(self):
         initial_channel_info = self.env['mail.channel'].with_user(test_user_1).with_context(allowed_company_ids=company_A.ids).channel_get(test_user_2.partner_id.ids)
         self.assertTrue(initial_channel_info, 'should be able to chat with multi company user')
 
+    def test_multi_company_message_post_notifications(self):
+        company_1 = self.env['res.company'].create({'name': 'Company 1'})
+        company_2 = self.env['res.company'].create({'name': 'Company 2'})
+        user_1 = self.env['res.users'].create({
```

Comment:
Use helper to create users + re-use existing users wen possible (for first company we may use users defined in common).